tests/components/gree/test_bridge.py

```diff
@@ -69,3 +75,33 @@ async def test_discovery_after_setup(
     device_infos = [x.device.device_info for x in hass.data[GREE][COORDINATORS]]
     assert device_infos[0].ip == "1.1.1.2"
     assert device_infos[1].ip == "2.2.2.1"
+
+
+async def test_coordinator_updates(
+    hass: HomeAssistant, freezer: FrozenDateTimeFactory, discovery, device, mock_now
+) -> None:
+    """Test gree devices update their state."""
+    await async_setup_gree(hass)
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all(DOMAIN)) == 1
+
+    coordinator = hass.data[GREE][COORDINATORS][0]
+
+    def update_device_state():
+        """Update the device state."""
+        coordinator.device_state_updated(["test"])
+
+    device().update_state.side_effect = update_device_state
+
+    next_update = mock_now + timedelta(seconds=UPDATE_INTERVAL)
+    freezer.move_to(next_update)
+
+    with patch.object(
+        DeviceDataUpdateCoordinator, "async_set_updated_data"
+    ) as mock_set_updated_data:
+        async_fire_time_changed(hass, next_update)
+        await hass.async_block_till_done()
+        mock_set_updated_data.assert_called_once_with(device().raw_properties)
+
+    assert coordinator.last_update_success is not None
```

Comment:
We should not touch the internals (`hass.data`) nor patch the the data update coordinator. We should rather get the callback function we registered from the mock. An example of this can be found in `knocki` tests

Reply:
It's unrelated to this change, shouldn't it be in another CL per submission policy?